src/psyclone/psyGen.py

```diff
@@ -142,19 +142,21 @@ def args_filter(arg_list, arg_types=None, arg_accesses=None, arg_meshes=None,
     arg_types and with access in arg_accesses. If these are not set
     then return all arguments.
 
-    :param arg_list: List of kernel arguments to filter
-    :type arg_list: list of :py:class:`psyclone.parse.algorithm.Descriptor`
-    :param arg_types: List of argument types (e.g. "GH_FIELD")
+    :param arg_list: list of kernel arguments to filter.
+    :type arg_list: list of :py:class:`psyclone.parse.kernel.Descriptor`
+    :param arg_types: list of argument types (e.g. "GH_FIELD").
```

Comment:
If we moved to sub-classing `Argument` (see discussion elsewhere), this check would just become `isinstance`.

Reply:
Agreed. Created #859 to discuss sub-classing `Argument`.